### PR TITLE
Devenv: self-instrumentation

### PR DESCRIPTION
**Pull request by LudoVio**
**What is this feature?**

Add a devenv source for instrumentation of Grafana itself

**Why do we need this feature?**

It is hard to work with Grafana instrumentation locally, create and explore metric, log and trace on both backend and frontend

**Who is this feature for?**

Every contributor to Grafana

### CODE REVIEW

File: devenv/docker/blocks/self-instrumentation/readme.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	apiUserTotal.Inc();
+
+	return hs.getUserUserProfile(c, c.UserID)
+}
+```
+
+
+You can now visit your [profile](http://localhost:3000/profile) page a few times and run a query on the Prometheus datasource to see your custom metric:
+
+http://localhost:3000/explore?orgId=1&left=%7B%22datasource%22:%22e34584fb-98e4-4d5c-83ae-3c0000dea48a%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22grafana_my_custom_api_user_total%7B%7D%22,%22range%22:true,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22e34584fb-98e4-4d5c-83ae-3c0000dea48a%22%7D,%22editorMode%22:%22code%22%7D%5D,%22range%22:%7B%22from%22:%22now-5m%22,%22to%22:%22now%22%7D%7D
```
Comment (gtk-grafana) [inline devenv/docker/blocks/self-instrumentation/readme.md:83]: this datasource ID won't exist for anyone else... unless we provision these datasources first!

In [setup.sh](https://github.com/grafana/grafana/blob/main/devenv/setup.sh#L46-L49) it looks like in the current devenv setup we're provisioning by creating `/conf/provisioning/datasources/dev.yaml`
Can we provision these new datasources as well?

File: devenv/docker/blocks/self-instrumentation/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - ../data/log:/var/log/grafana
+  tempo:
+    image: grafana/tempo:latest
+    command: -config.file=/etc/tempo.yaml
+    volumes:
+      - ./docker/blocks/self-instrumentation/tempo.yaml:/etc/tempo.yaml
+    ports:
+      - "14268:14268"
+      - "3200:3200"
+  # agent:
```
[the remark this answers is not in the document]
Comment (gtk-grafana) [reply devenv/docker/blocks/self-instrumentation/docker-compose.yaml:28]: I added this back in and it's working for me to test Faro => loki => grafana

File: devenv/docker/blocks/self-instrumentation/readme.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+This will setup Prometheus, Loki and Tempo.
+
+You then need to run Grafana with those added config:
+
+```ini
+[log.frontend]
+provider = grafana
+custom_endpoint = http://localhost:12347/collect
+api_key = api_key
```
Comment (gtk-grafana) [inline devenv/docker/blocks/self-instrumentation/readme.md:17]: Is this api_key being used for anything in this PR @LudoVio ?

File: devenv/docker/blocks/self-instrumentation/readme.md (hunk 1 above)
Comment (LudoVio) [reply devenv/docker/blocks/self-instrumentation/readme.md]: removed this

File: devenv/docker/blocks/self-instrumentation/docker-compose.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (LudoVio) [reply devenv/docker/blocks/self-instrumentation/docker-compose.yaml]: I put it back. Also you can now have frontend log without using agent

**Review by itsmylife — APPROVED** [review]

**Pull request merged**